File: ddd-application/src/main/java/com/zhuzichu/application/assembler/UserAssembler.java
package com.zhuzichu.application.assembler;


import com.zhuzichu.application.vo.UserVo;
import com.zhuzichu.domain.entity.User;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface UserAssembler {

    UserAssembler INSTANCE = Mappers.getMapper(UserAssembler.class);

    UserVo toUserVo(User user);

}

File: ddd-application/src/main/java/com/zhuzichu/application/assembler/UserAssembler.kt
package com.zhuzichu.application.assembler

import com.zhuzichu.application.vo.UserVo
import com.zhuzichu.domain.entity.User
import javax.inject.Inject


class UserAssembler @Inject constructor() {

    fun toUserVo(user: User): UserVo {
        return UserVo(user.nickname)
    }


}